src/plugins/discover/server/plugin.ts

```diff
@@ -56,6 +59,10 @@ export class DiscoverServerPlugin
 
     plugins.embeddable.registerEmbeddableFactory(createSearchEmbeddableFactory());
 
+    plugins.contentManagement.favorites.registerFavoriteType('esql_query', {
+      typeMetadataSchema: schema.object({ query: schema.string() }),
```

Comment:
TODO: finialize this + move to a separate file? Update the API tests when schema is finilized

Reply:
@Dosant this service is going to be used by all the places where the editor is and not only from Discover. I think we should move this registration to the esql plugin instead